### ISSUE TITLE
Cannot find module react-dom/server

### ISSUE THREAD

**Issue by MartinMuzatko** [asyncapi/html-template#228]
#### Describe the bug
I get an error of missing modules. 

I'm running on 
node v16.3.0
npm v7.15.1
Ubuntu 18.04.5 LTS (WSL 2)

#### How to Reproduce
`sudo ag https://raw.githubusercontent.com/GPS-GmbH/mqtt-spec/main/smqtt-public-1.0.yml @asyncapi/html-template -o mqtt`
![image](https://user-images.githubusercontent.com/2950505/124153381-a567f280-da94-11eb-8aeb-0e1e151eeff2.png)

```
> $ sudo ag https://raw.githubusercontent.com/GPS-GmbH/mqtt-spec/main/smqtt-public-1.0.yml @asyncapi/html-template -o mqtt
Something went wrong:
Error: Cannot find module 'react-dom/server'
Require stack:
- /usr/lib/node_modules/@asyncapi/generator/node_modules/@asyncapi/html-template/filters/all.js
- /usr/lib/node_modules/@asyncapi/generator/lib/filtersRegistry.js
- /usr/lib/node_modules/@asyncapi/generator/lib/generator.js
- /usr/lib/node_modules/@asyncapi/generator/cli.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:941:15)
    at Function.Module._load (node:internal/modules/cjs/loader:774:27)
    at Module.require (node:internal/modules/cjs/loader:1013:19)
    at require (node:internal/modules/cjs/helpers:93:18)
    at Object.<anonymous> (/usr/lib/node_modules/@asyncapi/generator/node_modules/@asyncapi/html-template/filters/all.js:4:24)
    at Module._compile (node:internal/modules/cjs/loader:1109:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1138:10)
    at Module.load (node:internal/modules/cjs/loader:989:32)
    at Function.Module._load (node:internal/modules/cjs/loader:829:14)
    at Module.require (node:internal/modules/cjs/loader:1013:19)
```

#### Expected behavior
It should install all required dependencies on its own.

**Comment by magicmatatjahu** [comment]
@MartinMuzatko Hello! I tried to your example and yes we have some problem as I see, but... If I choose manually the version of template (`0.23.0` which is latest version) using the command:

```bash
ag https://raw.githubusercontent.com/GPS-GmbH/mqtt-spec/main/smqtt-public-1.0.yml @asyncapi/html-template@0.23.0 -o mqtt
```

everything works as expected. Could you try in your machine? After successfully generating with `0.23.0` I don't need specify the version and I can use your command.

**Comment by MartinMuzatko** [comment]
thanks for moving it over! :)
Looks good, it works with the version 0.23.0 and after that I can omit the version.
Is this how it is supposed to be and I missed something in the documentation?  :D
In any other case:
Is npm not able to correctly resolve the latest version maybe? I cannot even start to comprehend how this issue exists.

**Comment by derberg** [comment]
@MartinMuzatko I recommend looking at [this](https://github.com/asyncapi/generator/#generator-version-vs-template-version) part of generator readme. 

tl;dr Generator is versioned independently from templates. Also, every new version of a template can introduce changes that you might not necessarily like. So, like with docker images or any other things, it is recommended to always use fixed version long term to not be surprised on production while using `latest` 😄 

> Looks good, it works with the version 0.23.0 and after that I can omit the version.

hard to say without detailed debugging, maybe you already had HTML template in your node_modules 🤷🏼 so it was just picking up that version. Generator explicitly tries to install package again only if you pass `--install` flag, otherwise it is trying to pick up cached version from node_modules. Hard to say if this was the reason though.

**Comment by magicmatatjahu** [comment]
As derberg said, it's hard to say what problem was there. If you had a problem with missed `react-dom` package there npm could install template but "skip" some dependencies to install like mentioned. I think that it could be a problem with installation of puppeter, which has >100mb weight. Sometimes I had similar errors related to to missed dep which had errors on installation, but npm saw that root package is "installed", so I had to reinstall the package from scratch...

We can suggest to use the fixed version or always pass the `PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=true` env - probably there could be a problem when npm tries to install the huge package and, with timeouts, skips the others, tbh, I don't know. 

Could we close the issue, or you have another questions @MartinMuzatko? :)

**Comment by MartinMuzatko** [comment]
Thank you for answering all my questions so swiftly. We can close this issue :)
Maybe it would help to update the command presented on https://www.asyncapi.com/generator to always append the latest version of the `html-template`. But maybe that is too much overhead.

**Comment by magicmatatjahu** [comment]
Good point, we can use the `latest` tag like `@asyncapi/html-template@latest` and we don't need to define the numeric latest version  :)

Ok, if someone will have similar problem, please reopen the issue.
